esp/scm/ws_workunits.ecm

```diff
@@ -1187,11 +1187,13 @@ ESPrequest WUQueryGetSummaryStatsRequest
     string QueryId;
     string FromTime; //YYYY-MM-DDTHH:MM:SS
     string ToTime;
+    [min_ver("1.75")] bool All(false); //Return an array of all the raw stats that are available in the nominated time frame
 };
 
 ESPresponse [exceptions_inline] WUQueryGetSummaryStatsResponse
 {
     ESParray<ESPstruct QuerySummaryStats> StatsList;
+    [min_ver("1.75")] string RawStats;
```

Comment:
I was expecting the raw stats to be of the same shape as a `QuerySummaryStats`?

Reply:
I thought you might say that.  So All would just be a way of removing all the filters, but the info would still be returned the same way.

Reply:
Yes

Reply:
@wangkx - at the moment if you pass in no From/To you get 1 stat back with the entire date range.
If "All" is true I would expect an array of all the stats back (in the same format).

Maybe it might be better if we called it "Aggregate" and defaulted it on?  That way I can ask for a time range and get the aggregated result (singular) or the non - aggregated results (plural)?

Reply:
Correct me if I do not understand correctly:
1. Change the "All" to "Aggregate".
2. default "Aggregate" to on. 
3. When it is on, the WUQueryGetSummaryStats returns as what we have before this PR.
4. When it is not on,  the WUQueryGetSummaryStats returns the non aggregated results  for multiple time ranges.
Questions/concerns: 1. who defines the multiple time ranges? 2. Current QuerySummaryStats (one for each roxie node) already contains all of stats data items returned from roxie. What if some stats data items are added in future (that is why I am using the string RawStats here)? 
@GordonSmith

Reply:
1 - Roxie does
2 - Its not about new "types" of stats - its about new stat records for new time frames.

My understanding is that Roxie maintains stats for:
* every minute (for the last hour)
* every hour (for the last day)
* every day since the server came up

I would like to get that entire list - rather than 1 stat for the specified time frame?

Reply:
Ok, this PR is about new stat records for new time frames.

The response structure may be changed:
ESPStruct QuerySummaryStatsInATimeFrame
{
    string From;
   string To;
   ESParray<ESPstruct QuerySummaryStats> Stats;
};

ESPresponse [exceptions_inline] WUQueryGetSummaryStatsResponse
{
    ESParray<ESPstruct QuerySummaryStatsInATimeFrame> Stats;
};

I need to find out how to get that entire list (hopefully in one roxie control call).

Reply:
@richardkchapman which roxie <control> call can be used to get all (not aggregated) stat records in all time frames for a roxie query (please see Gordon's comment above)?

Reply:
They are aggregated as they get less current, for reasons of space. Gordon wants all available information, not all information that has ever existed...

Reply:
@richardkchapman could you please tell me how to get the 'all available information'?

Reply:
else if (stricmp(queryName, "control:queryAggregates")==0)
            {
                time_t from;
                const char *fromTime = control->queryProp("@from");
                if (fromTime)
                {
                    CDateTime f;
                    f.setString(fromTime, NULL, true);
                    from = f.getSimple();
                }
                else
                    from = startupTime;
                time_t to;
                const char *toTime = control->queryProp("@to");
                if (toTime)
                {
                    CDateTime t;
                    t.setString(toTime, NULL, true);
                    to = t.getSimple();
                }
                else
                    time(&to);
                const char *id = control->queryProp("Query/@id");
                if (id)
                {
                    Owned<IQueryFactory> f = getQuery(id, NULL, NULL, logctx);
                    if (f)
                    {
                        Owned<const IPropertyTree> stats = f->getQueryStats(from, to);
                        toXML(stats, reply);
                    }
                    else
                        throw MakeStringException(ROXIE_CONTROL_MSG_ERROR, "Unknown query %s", id);
                }
                else
                {
                    bool includeAllQueries = control->getPropBool("@all", true);
                    Owned<const IPropertyTree> stats = getAllQueryStats(includeAllQueries, from, to);
                    toXML(stats, reply);
                }
            }